### PR TITLE
framework/wifi_manager: Distinguish RECONNECT and DISCONNECT when callback

### PR DESCRIPTION
**Pull request by SeonghoByeon**
- Pass an additional parameter, wifi_manager_disconnect_to, to disconnect callback
  in order for the callback function to distinguish
  whether Wi-Fi Manager goes into RE(AUTO)CONNECT or not
- Add enum, wifi_manager_disconnect_to,  to distinguish RECONNECT and DISCONNECT
  when calling disconnect callback
- Modify the location of link up/down messages passed to iotivity
- Modify all of the sta_disconnected callback functions used in the applications

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [b318f6304e421d751cde19b4fd47caece30ddef7] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b318f6304e421d751cde19b4fd47caece30ddef7] - Code Rule Check (C++) OK.

File: framework/include/wifi_manager/wifi_manager.h [hunk 1]
```diff
@@ -116,7 +124,7 @@ typedef struct wifi_manager_scan_info_s wifi_manager_scan_info_s;
  */
 typedef struct {
 	void (*sta_connected)(wifi_manager_result_e);	// in station mode, connected to ap
-	void (*sta_disconnected)(void);		// in station mode, disconnected from ap
+	void (*sta_disconnected)(wifi_manager_disconnect_to);		// in station mode, disconnected from ap
```
Comment (JeongJunSik) [inline framework/include/wifi_manager/wifi_manager.h]: If the public callback API changes, how about change it all together in tizenRT?

Comment (SeonghoByeon) [reply framework/include/wifi_manager/wifi_manager.h]: Yes, we have decided to change the public callback API of Wi-Fi Manager to provide the information of re(auto)connect. Based on the modification, I changed all corresponding disconnect callback function in various apps and framework such as st_things.

Comment (JeongJunSik) [reply framework/include/wifi_manager/wifi_manager.h]: Oh yes,
I have not seen the second commit.

File: framework/include/wifi_manager/wifi_manager.h [hunk 2]
```diff
@@ -67,6 +67,14 @@ typedef enum {
 	WIFI_MANAGER_ALREADY_CONNECTED,
 } wifi_manager_result_e;
 
+/**
+ * @brief Wi-Fi disconnect event reason
+ */
+typedef enum {
+	WIFI_MANAGER_DISCONNECT,
+	WIFI_MANAGER_RECONNECT, //AUTOCONNECT
+} wifi_manager_disconnect_to;
```
Comment (pillip8282) [inline framework/include/wifi_manager/wifi_manager.h]: you have to use _e instead of _to; it's conventional for example wifi_manager_result_e

**Comment by seinfra** [comment]
Target : [f563c4977ab6fbd82456c20ce83d5825788d1ed8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f563c4977ab6fbd82456c20ce83d5825788d1ed8] - Code Rule Check OK.

File: framework/include/wifi_manager/wifi_manager.h (hunk 2 above)
Comment (SeonghoByeon) [reply framework/include/wifi_manager/wifi_manager.h]: Thanks you for informing. The code now follows the convention.
Modification completed.

**Pull request merged**